app/browser/tabs.js

```diff
@@ -985,6 +988,86 @@ const api = {
 
     return nextTabId
   },
+
+  closeTabPage: (state, windowId, tabPageIndex) => {
+    const tabsPerPage = Number(getSetting(settings.TABS_PER_PAGE))
+    const startTabIndex = tabPageIndex * tabsPerPage
+    const pinnedTabsCount = tabState.getPinnedTabsByWindowId(state, windowId).size
+    tabState.getTabsByWindowId(state, windowId)
```

Comment:
There's a whole bunch happening here- would it make sense to follow up and move this logic into tabState? Like a tabState.getTabsForClose (or similar)

Reply:
I think it makes sense but I'm going to pass on it for now since I have other 0.19.x high priority things to focus on, but that's valid minor refactor for sure.